tests/integration/test_dev_server.py

```diff
@@ -0,0 +1,156 @@
+import asyncio
+import nikola.plugins.command.auto as auto
+from nikola.utils import get_logger
+import pytest
+import pathlib
+import socket
+from typing import Optional
+import urllib.request
+
+from ..helper import FakeSite
+
+SERVER_ADDRESS = "localhost"
+TEST_MAX_DURATION = 10  # Watchdog: Give up the test if it did not succeed during this time span.
+SERVER_STARTUP_TIME = 0.1  # Max time, in seconds, we expect the server to need to come up.
```

Comment:
1. This seems to be unused.
2. The value feels a bit too optimistic.

Reply:
That's a leftover that needs to be deleted.

TMI unless you are interested: I tried to schedule the test proper within the loop, with this much delay. But then, the blocking IO of the HTTP client blocks the thread, so the loop to no longer makes progress, so the server doesn't service the incoming HTTP request, so this whole thing yams. Running that HTTP client via `loop.run_in_executor` fixes that problem.

(The server startup time here is much faster than what we are used to, as there is no build happening here. It is really only HTTP server setup.)

Reply:
When writing tests that measure time, it’s always good to have some margin to avoid flakiness. The tests may end up running on a slower machine that’s busy with multiple concurrent processes (as CI servers are). For a timeout like this, I’d use 1-2 seconds to be safe.